arkouda_viz/client/arkouda_viz/viz.py

```diff
@@ -0,0 +1,140 @@
+"""Module providing our plotting capabilities."""
+import holoviews as hv
+import panel as pn
+import panel.widgets as pnw
+from arkouda.dataframe import DataFrame
+from typeguard import typechecked
+from holoviews.operation import datashader as ds
+
+pn.extension()
+
+
+def datashade(
+    data: DataFrame,
+    width: int = 500,
+    height: int = 500,
+    engine: str = "bokeh",
+    point_size: int = 1,
+):
+    """Creates an interactive plot that can handle large datasets."""
+    hv.extension(engine)
+
+    data = data.to_pandas().select_dtypes(exclude=["object"])
+
```

Comment:
just pointing out for really large datasets, this isn't really feasible because `to_pandas` pulls all the data from the server onto the client. This is why we have `ak.client.maxTransferBytes`

Reply:
Yeah, I will keep it for now, but the plan is to move more onto the server.

Reply:
Sounds good!! Feel free to reach out if u need any help writing chpl code to do that